File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetscommerceproductpagedresponsemodel.go
package shared

// CodatDataContractsDatasetsCommerceProductPagedResponseModel
// Used to represent what can be returned by an endpoint that supports paging.
// Usable with the [ProducesResponseType] attribute on a controller action.
type CodatDataContractsDatasetsCommerceProductPagedResponseModel struct {
	Links        *CodatDataContractsDatasetsCommerceProductPagedResponseLinksModel `json:"_links,omitempty"`
	PageNumber   *int                                                              `json:"pageNumber,omitempty"`
	PageSize     *int                                                              `json:"pageSize,omitempty"`
	Results      []CodatDataContractsDatasetsCommerceProduct                       `json:"results,omitempty"`
	TotalResults *int                                                              `json:"totalResults,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetscommerceproductpagedresponsemodel.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { CodatDataContractsDatasetsCommerceProductPagedResponseLinksModel } from "./codatdatacontractsdatasetscommerceproductpagedresponselinksmodel";
import { CodatDataContractsDatasetsCommerceProduct } from "./codatdatacontractsdatasetscommerceproduct";



// CodatDataContractsDatasetsCommerceProductPagedResponseModel
/** 
 * Used to represent what can be returned by an endpoint that supports paging.
 * Usable with the [ProducesResponseType] attribute on a controller action.
**/
export class CodatDataContractsDatasetsCommerceProductPagedResponseModel extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=_links" })
  links?: CodatDataContractsDatasetsCommerceProductPagedResponseLinksModel;

  @SpeakeasyMetadata({ data: "json, name=pageNumber" })
  pageNumber?: number;

  @SpeakeasyMetadata({ data: "json, name=pageSize" })
  pageSize?: number;

  @SpeakeasyMetadata({ data: "json, name=results", elemType: CodatDataContractsDatasetsCommerceProduct })
  results?: CodatDataContractsDatasetsCommerceProduct[];

  @SpeakeasyMetadata({ data: "json, name=totalResults" })
  totalResults?: number;
}
